sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -17,40 +17,264 @@ namespace Azure.AI.OpenAI
     /// <summary> Azure OpenAI APIs for completions and search. </summary>
     public partial class OpenAIClient
     {
+        private const string publicOpenAIVersion = "1";
+        private const string publicOpenAIEndpoint = $"https://api.openai.com/v{publicOpenAIVersion}";
+
+        /// <remarks> This is the deployment id for all Azure OpenAI operations (completions and embeddings) </remarks>
+        private string DeploymentId { get; }
+
+        /// <remarks> This key is used to connect to a non-Azure OpenAI resource created directly with OpenAI.
+        /// For Azure OpenAI resources as created and maintained in Azure Portal, use a constructor that provides
+        /// an Azure resource endpoint and credential, instead. </remarks>
+        private string PublicOpenAIToken { get; }
+
+        /// <summary> Initializes a instance of OpenAIClient using the public OpenAI endpoint. </summary>
+        /// <param name="openAIAuthToken "> String token to generate a token credential.
+        /// This string is used to connect to a non-Azure OpenAI resource created directly with OpenAI.
+        /// For Azure OpenAI resources as created and maintained in Azure Portal, use a constructor that provides
+        /// an Azure resource endpoint and credential, instead. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="openAIAuthToken"/> is null. </exception>
+        public OpenAIClient(string openAIAuthToken) : this(openAIAuthToken, new OpenAIClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a instance of OpenAIClient using the public OpenAI endpoint. </summary>
+        /// <param name="openAIAuthToken "> String token to generate a token credential.
+        /// This string is used to connect to a non-Azure OpenAI resource created directly with OpenAI.
+        /// For Azure OpenAI resources as created and maintained in Azure Portal, use a constructor that provides
+        /// an Azure resource endpoint and credential, instead. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="openAIAuthToken"/> is null. </exception>
+        public OpenAIClient(string openAIAuthToken, OpenAIClientOptions options) : this(new Uri(publicOpenAIEndpoint), "", CreateDelegatedToken(openAIAuthToken), options)
```

Comment:
Ah, I see! Then, in that case, I would recommend passing `null` instead of an empty string. That being said, it's okay to keep it like this for now since the `DeploymentId` is currently private anyway, and we can wait until the TypeSpec is updated to revisit this constructor.